### PR TITLE
[l10n-action] children should not be considered when measuring space (fixes #42)

### LINKED ISSUE #42: l10n-action nodes inside the header should not be measured
This causes the title text to be incorrectly aligned.

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

File: gaia-header.js [hunk 1]
```diff
@@ -519,6 +519,7 @@ module.exports = component.register('gaia-header', {
    * @private
    */
   getButtonsBeforeTitle: function() {
+    var tags = { a: 1, button: 1 };
```
Comment (julienw) [inline gaia-header.js]: is it a left over ?

File: gaia-header.js [hunk 2]
```diff
@@ -967,10 +968,16 @@ function dir() {
  * Determines whether passed element
  * contributes to the layout in gaia-header.
  *
+ * Children with `[l10n-action]` get distributed
+ * inside the action-button so don't occupy
+ * space before the `<h1>`.
```
Comment (julienw) [inline gaia-header.js]: isn't the right reason that it's "position: absolute" ?

File: README.md [hunk 3]
```diff
@@ -46,11 +46,11 @@ Prevents font-fit logic from running. Will run when the attribute is removed. Yo
 
 ### Localization
 
-If choosing to use the built in action-button you may wish to localize the content within. You can do this by providing a specicial `.l10n-action` child node.
+If choosing to use the built in action-button you may wish to localize the content within. You can do this by providing a specicial `l10n-action` child node. This node will be 'distributed' inside the action-button acting as `textContent` for screen-readers.
```
Comment (julienw) [inline README.md]: nit: please fix the typo "specicial".

File: gaia-header.js (hunk 2 above)
Comment (julienw) [inline gaia-header.js]: ok it's actually both, forget my comment then :)

File: gaia-header.js (hunk 1 above)
Comment (julienw) [inline gaia-header.js]: and that's why travis is failing :)

File: test/test-unit.js [hunk 4]
```diff
@@ -355,6 +354,17 @@ suite('GaiaHeader', function() {
...
+    test('children with .l10n-action are not considered when calculating title space', function() {
+      this.dom.innerHTML = '<gaia-header action="back"><span l10n-action>' +
+        '</span><h1>Title title</h1></gaia-header>';
+      var header = this.dom.firstElementChild;
+      var h1 = header.querySelector('h1');
+
+      return afterNext(header, 'runFontFit').then(() => {
+        assert.equal(h1.style.marginInlineStart, '-50px');
+      });
+    });
```
Comment (julienw) [inline test/test-unit.js]: mmm this test is not failing on master. Can you do a test failing in master, but passing with your patch ?

**Comment by julienw** [comment]
Mostly nits, but I'd also like a test that fails in master ;)

Thanks

File: test/test-unit.js (hunk 4 above)
Comment (wilsonpage) [inline test/test-unit.js]: Fixed

**Comment by wilsonpage** [comment]
Fixed all comments, waiting for green :+1:

**Pull request merged**